### ISSUE TITLE
Custom file name pattern

### ISSUE THREAD

**Issue by chr1s1k** [ionic-team/stencil#2803]
**Stencil version:**
```
 @stencil/core@2.3.0
```

**I'm submitting a:**

[ ] bug report
[x] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

Hi, is there a possibility to configure the name of generated files?

**Example of currently generated files:**
```
namespace.js
namespace.esm.js
p-12345.entry.js
```

**Example of custom file name pattern:**
```
namespace.js
namespace-esm.js
p-12345-entry.js
```

Basically, I would like to replace all the dots within the file name except the last one.

**Comment by claviska** [comment]
This isn't currently possible as the extensions are hard-coded:

```ts
const entryEsmFilename = `${config.fsNamespace}.esm.js`;
const entryNoModuleFilename = `${config.fsNamespace}.js`;
```

Is the extra dot causing some kind of issue?

**Comment by chr1s1k** [comment]
Thanks for letting me know.

Unfortunately, I'm using a CMS which can't allow me to deploy JS files with more than one dot :) I know it sounds very strange but right now I'm in the middle of an investigation process if I could somehow bypass this CMS restriction.

I've been also thinking about some post-build task which can go through all the generated files, modify the names and replace all occurrences of such names in those files but I know this would be very unusual and extensive operation.

**Comment by xtrnavsky** [comment]
@chr1s1k and what about deploy components to some CDN and reference from there?

**Comment by erichstark** [comment]
@chr1s1k and what about deploy components to some CDN and reference from there?
